src/shutdown.h

```diff
@@ -19,6 +19,9 @@ bool InitShutdownState();
 /** Request shutdown of the application. */
 void StartShutdown();
 
+/** Request application shutdown due an internal error. */
+void StartErrorShutdown();
```

Comment:
The reason behind that is https://github.com/bitcoin/bitcoin/pull/27607#discussion_r1199624849.
But.. could not expose it and move the caller to use `AbortNode()` instead. Same as we do with the external block import failures.

Reply:
Still, in order to do that in the cleanest possible way, will need to refactor `shutdown.cpp`, so `AbortNode` is placed after the static `StartErrorShutdown` function. Which I'm not so sure that worth it due the conflicts that could cause with your PRs. Thoughts?

Reply:
So IIRC, this call [here](https://github.com/bitcoin/bitcoin/commit/ca3041984cf3665e27b6783c923ab1c32682500a#diff-b1e19192258d83199d8adaa5ac31f067af98f63554bfdd679bd8e8073815e69dR1669) would call `StartErrorShutdown`?

Reply:
> Which.. isn't bad as it will remove another ui_interface dependency.

Can you elaborate on this point? `AbortNode` still relies on `ui_interface` from what I can tell. 

From my current understanding both the `StartShutdown` in `ThreadImport` and in the check are call shutdown and node `AbortNode`, because `BlockImport` might run before the ui is available (this is me guessing out of historical reasons). I am not sold on this point though, since this runs in another thread and we also provide the noui consumer of the ui interface. So I think these should both become calls to `AbortNode`.

Reply:
> > Which.. isn't bad as it will remove another ui_interface dependency.
> 
> Can you elaborate on this point? `AbortNode` still relies on `ui_interface` from what I can tell.

Ok, I confused you. Sorry.

The ui_interface dependency that can be removed is the `base.cpp` one. Thus why said to "bubble up" the error string up to `init.cpp` instead of calling `InitError` from the index base class `Start` method.

> From my current understanding both the `StartShutdown` in `ThreadImport` and in the check are call shutdown and node `AbortNode`, because `BlockImport` might run before the ui is available (this is me guessing out of historical reasons).

`AbortNode` isn't doing anything special. We already use `InitError` in `init.cpp` prior spawning the import thread.
What the node will do when there is no gui is print the error to the command line, and the debug file, prior shutting down (using the noui consumer as you said).

> I am not sold on this point though, since this runs in another thread and we also provide the noui consumer of the ui interface. So I think these should both become calls to `AbortNode`.

Yeah, that is my idea as well. The `ThreadImport` "fatal" failures should use `AbortNode`.